core/state/state_object.go

```diff
@@ -120,15 +197,22 @@ func newObject(db *StateDB, address common.Address, data Account) *StateObject {
 	if data.Root == (common.Hash{}) {
 		data.Root = emptyRoot
 	}
-	return &StateObject{
-		db:             db,
-		address:        address,
-		addrHash:       crypto.Keccak256Hash(address[:]),
-		data:           data,
-		originStorage:  make(Storage),
-		pendingStorage: make(Storage),
-		dirtyStorage:   make(Storage),
+	s := &StateObject{
+		db:         db,
+		address:    address,
+		addrHash:   crypto.Keccak256Hash(address[:]),
+		data:       data,
+		isParallel: isParallel,
+	}
+	s.originStorage, s.dirtyStorage, s.pendingStorage = s.newStorage(), s.newStorage(), s.newStorage()
```

Comment:
move it back to `StateObject{}`?
`StateObject{`
`...`
`originStorage: s.newStorage()`
`...`
`}`